staging/src/k8s.io/kube-aggregator/pkg/apiserver/openapi_aggregator.go

```diff
@@ -47,36 +48,65 @@ type openAPIAggregator struct {
 	// provided for dynamic OpenAPI spec
 	openAPIService *handler.OpenAPIService
 
-	// Aggregator's OpenAPI spec (holds apiregistration group).
-	aggregatorOpenAPISpec *spec.Swagger
+	contextMapper request.RequestContextMapper
+}
 
-	// Local (in process) delegate's OpenAPI spec.
-	inProcessDelegatesOpenAPISpec *spec.Swagger
+func (s *openAPIAggregator) LoadAndAddLocalSpec(localHandler http.Handler, name string, filterPaths []string) error {
+	spec, err := s.downloadOpenAPISpec(localHandler)
+	if err != nil {
+		return err
+	}
+	if filterPaths == nil {
+		filterPaths = []string{"/apis/"}
+	}
+	if len(filterPaths) > 0 {
+		aggregator.FilterSpecByPaths(spec, filterPaths)
+	}
+	s.addLocalSpec(spec, name)
+	return nil
+}
 
-	contextMapper request.RequestContextMapper
+func (s *openAPIAggregator) addLocalSpec(spec *spec.Swagger, name string) {
+	localApiService := apiregistration.APIService{}
+	localApiService.Name = name
```

Comment:
so this name is not actually the one from the auto registration controller of the aggregator, but some artifical one per delegation apiserver in the chain?